pkg/cache/placeholder_manager.go

```diff
@@ -70,10 +70,28 @@ func (mgr *PlaceholderManager) createAppPlaceholders(app *Application) error {
 	return nil
 }
 
-// recycle all the placeholders for an application
-func (mgr *PlaceholderManager) Recycle(appID string) {
-	log.Logger().Info("start to recycle app placeholders",
-		zap.String("appID", appID))
+// clean up all the placeholders for an application
+func (mgr *PlaceholderManager) CleanUp(app *Application) {
+	log.Logger().Info("start to clean up app placeholders",
+		zap.String("appID", app.GetApplicationID()))
+	for taskID, task := range app.taskMap {
+		if task.GetTaskPlaceholder() {
+			// remove pod
+			err := mgr.clients.KubeClient.Delete(task.pod)
+			if err != nil {
+				log.Logger().Error("failed to clean up placeholder pod",
+					zap.Error(err))
```

Comment:
When a deletion fails, if we do not do anything, this will create an "orphan" pod, which will leak some resources on the cluster, no good. I've created a JIRA: https://issues.apache.org/jira/browse/YUNIKORN-469 to handle such failures.

Reply:
OK, I have add `orphanpod` map in placeholder manager. 
And leverage it in YUNIKORN-469